dashboard/app/helpers/levels_helper.rb

```diff
@@ -316,7 +316,9 @@ def app_options
     end
 
     @app_options =
-      if @level.is_a? Blockly
+      if @script_level && @level.uses_lab2?
+        {app: 'lab2', channel: view_options[:channel]}
```

Comment:
Will we not use lab2 for non-script single levels (levels/xyz urls)?

Reply:
It's a good question.  The `Lab2` component right now has some assumptions that it's part of a `ScriptLevel`, since it examines the current lesson and has logic for loading multiple apps at once.  Maybe it should be renamed to `Lab2ScriptLevel` or similar?  

I'm definitely open to more comprehensive refactoring, but this change is also getting big enough that I wouldn't mind getting it merged, and then we can do more refactoring as follow-ups.  

What do you think?

Reply:
Gotcha - yeah I think it would be nice to reduce/remove the dependency on progress redux so that Lab2 can be used for any lab on any page, but agreed that it can happen as a follow up. From what i'm seeing, I think if we were to pass the app type in as part of level properties, we shouldn't need to look up app type on the lesson, right? As far as loading multiple app views - it would be interesting to see how many apps need to stay rendered vs how many can be unmounted, but if it there aren't many that need to stay rendered, I think we can get away with handling multiple renders for all app types, rather than ones just in the lesson.

Reply:
> From what i'm seeing, I think if we were to pass the app type in as part of level properties, we shouldn't need to look up app type on the lesson, right?

While we'd know the app type of the current level, we'd still want to know the app type of the destination level ahead of time, so that we can switch immediately to it and then initiate the retrieval of its own level properties.

Reply:
Right yeah, but that's only for level switching right? Within the Lab2 container, would it be feasible to only worry about rendering the current app? In the case of unmountable apps like music, we can have some logic that only renders them the first time we encounter them, and then keeps them hidden on the page.

Again, totally fine with this approach, just want to see how flexible we can make this

Reply:
Oh, that's an interesting idea.  I guess the current technique effectively "preloads" Music Lab so that it's already in the DOM, just not visible, but with this idea we are more opportunistic about what we load.  Makes sense, and worth exploring!